File: angularapp/src/app/services/property-inquiry.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Global } from '../global';
import { PropertyInquiry } from '../models/property-inquiry.model';
import { Observable } from 'rxjs';
import { User } from '../models/user.model';

@Injectable({
  providedIn: 'root'
})
export class PropertyInquiryService {
  
  constructor(private http :HttpClient) { }

  public addInquiry(inquiry :PropertyInquiry): Observable<PropertyInquiry>{
     return this.http.post(Global.baseUrl+"/api/inquiries",inquiry) as Observable<PropertyInquiry>;
  }

  public getAllInquiries() :Observable<PropertyInquiry[]>{
    return this.http.get(Global.baseUrl+"/api/inquiries") as Observable<PropertyInquiry[]>;
  }

  public getInquiriesByUserId(userId :number) :Observable<PropertyInquiry[]>{
     return this.http.get(Global.baseUrl+"/api/inquiries/user/"+userId) as Observable<PropertyInquiry[]>;
  }

  public updateInquiry(inquiryId: number,inquiry: PropertyInquiry) :Observable<PropertyInquiry>{
    return this.http.put(Global.baseUrl+"/api/inquiries/"+inquiryId,inquiry) as Observable<PropertyInquiry>;
  }

  public deleteInquiry(inquiryId :number) :Observable<any>{
    return this.http.delete(Global.baseUrl+"/api/inquiries/"+inquiryId) as Observable<any>;
  }

}

File: springapp/src/main/java/com/examly/springapp/service/PropertyInquiryService.java
package com.examly.springapp.service;
 
import java.util.List;
import java.util.Optional;
 
import com.examly.springapp.model.PropertyInquiry;
 
public interface PropertyInquiryService {
   
    public PropertyInquiry addInquiry(PropertyInquiry propertyInquiry);
    public Optional<PropertyInquiry> getInquiryById(long inquiryId);
    public List<PropertyInquiry> getAllInquiries();
    public PropertyInquiry updateInquiry(long inquiryId, PropertyInquiry propertyInquiry);
    public void deleteInquiry(long inquiryId);
    public List<PropertyInquiry> getInquiriesByUserId(long userId);
  
}
 
 
